components/server/src/code-sync/code-sync-service.ts

```diff
@@ -410,12 +571,34 @@ export class CodeSyncService {
         } catch (e) {
             if (rev) {
                 log.error({ userId }, "code sync: failed to delete obj", e, {
-                    object: toObjectName(resourceKey, rev),
+                    object: toObjectName(resourceKey, rev, collection),
                 });
             } else {
                 log.error({ userId }, "code sync: failed to delete", e);
             }
             throw e;
         }
     }
+
+    private async deleteCollection(userId: string, collection: string | undefined) {
+        try {
+            await this.db.deleteCollection(userId, collection, async (collection?: string) => {
+                try {
+                    const request = new DeleteRequest();
+                    request.setOwnerId(userId);
+                    request.setPrefix(toObjectName(SyncResource.GlobalState, "all", collection ?? "all"));
+                    const blobsClient = this.blobsProvider.getDefault();
+                    await util.promisify<DeleteRequest, DeleteResponse>(blobsClient.delete.bind(blobsClient))(request);
+                } catch (e) {
```

Comment:
We have duplicate code like this, why not make them inside this function? And it miss delete GCP (stored) resources too